tests/test_hubmixin.py

```diff
@@ -91,6 +94,57 @@ def test_save_pretrained(self):
         self.assertTrue("pytorch_model.bin" in files)
         self.assertEqual(len(files), 2)
 
+    def test_save_pretrained_with_push_to_hub(self):
+        REPO_NAME = repo_name("save")
+        save_directory = f"{WORKING_REPO_DIR}/{REPO_NAME}"
+        config = {"hello": "world"}
+        model = DummyModel()
+        model.push_to_hub = Mock()
+        model._save_pretrained = Mock()  # disable _save_pretrained to speed-up
+
+        # Not pushed to hub
+        model.save_pretrained(save_directory)
```

Comment:
If it's not much I would consider splitting the tests as they testing different behaviours

Reply:
See comment below.